tests/databases/test_redshift.py

```diff
@@ -0,0 +1,324 @@
+def test_table_exists_raises_exception():
+    """Test if table exists in redshift database"""
+    database = RedshiftDatabase(conn_id=CUSTOM_CONN_ID)
+    table = Table(name="inexistent-table", metadata=Metadata(schema=SCHEMA))
+    assert not database.table_exists(table)
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [
+        {
+            "database": Database.REDSHIFT,
+            "table": Table(
+                metadata=Metadata(schema=SCHEMA),
+                columns=[
+                    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+                    sqlalchemy.Column(
+                        "name", sqlalchemy.String(60), nullable=False, key="name"
+                    ),
+                ],
+            ),
+        }
+    ],
+    indirect=True,
+    ids=["redshift"],
+)
+def test_redshift_create_table_with_columns(database_table_fixture):
+    """Test table creation with columns data"""
+    database, table = database_table_fixture
+
+    statement = (
+        f"SELECT * FROM INFORMATION_SCHEMA.COLUMNS WHERE table_name='{table.name}'"
+    )
+    response = database.run_sql(statement)
+    assert response.first() is None
+
+    database.create_table(table)
+    response = database.run_sql(statement)
+    rows = response.fetchall()
+    assert len(rows) == 2
+    assert rows[0][0:4] == (
+        "dev",
+        SCHEMA,
+        f"{table.name}",
+        "id",
+    )
+    assert rows[1][0:4] == (
+        "dev",
+        SCHEMA,
+        f"{table.name}",
+        "name",
+    )
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [
+        {"database": Database.REDSHIFT},
+    ],
+    indirect=True,
+    ids=["redshift"],
+)
+def test_load_pandas_dataframe_to_table(database_table_fixture):
+    """Test load_pandas_dataframe_to_table against redshift"""
+    database, table = database_table_fixture
+
+    pandas_dataframe = pd.DataFrame(data={"id": [1, 2]})
+    database.load_pandas_dataframe_to_table(pandas_dataframe, table)
+
+    statement = f"SELECT * FROM {database.get_table_qualified_name(table)};"
+    response = database.run_sql(statement)
+
+    rows = response.fetchall()
+    assert len(rows) == 2
+    assert rows[0] == (1,)
+    assert rows[1] == (2,)
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [
+        {"database": Database.REDSHIFT},
+    ],
+    indirect=True,
+    ids=["redshift"],
+)
+def test_load_file_to_table(database_table_fixture):
+    """Test loading on files to redshift database"""
+    database, target_table = database_table_fixture
+    filepath = str(pathlib.Path(CWD.parent, "data/sample.csv"))
+    database.load_file_to_table(File(filepath), target_table, {})
+
+    df = database.hook.get_pandas_df(
+        f"SELECT * FROM {database.get_table_qualified_name(target_table)}"
+    )
+    assert len(df) == 3
+    expected = pd.DataFrame(
```

Comment:
@pankajkoti do we have guarantees of the ordering of rows? or should we explicitly sort rows?

Reply:
in past, I have seen test cases breaking because the order of rows changed.

Reply:
Sorted rows for all checks.